src/cursor/cur_hs.c

```diff
@@ -885,7 +919,7 @@ __curhs_insert(WT_CURSOR *cursor)
 
 #ifdef HAVE_DIAGNOSTIC
     /* Do a search again and call next to check the key order. */
-    WT_WITH_PAGE_INDEX(session, ret = __wt_hs_row_search(cbt, &file_cursor->key, true));
+    WT_WITH_PAGE_INDEX(session, ret = __curhs_search(cbt, &file_cursor->key, true));
```

Comment:
Since update and remove are always called with page pinned, there is no need to handle the restart case? @keithbostic

Reply:
I think you still need a restart handler, the modify routine is the one that's likely to return WT_RESTART because inserts into a  skip list raced.